### PR TITLE
feat: tricklenode

### PR DESCRIPTION
**Pull request by docmerlin**
This creates a node that converts from a batch to a stream, it could be thought of as the inverse of windownode.

### CODE REVIEW

File: pipeline/tick/tick_test.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
+
```
Comment (danxmoran) [inline pipeline/tick/tick_test.go:12]: Needs reformatted

File: trickle.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+type TrickleNode struct {
+	node
+	dims models.Dimensions
+	name string
+}
+
+func (n *TrickleNode) runTrickle(_ []byte) error {
+	consumer := edge.NewConsumerWithReceiver(
+		n.ins[0],
```
Comment (danxmoran) [inline trickle.go:25]: Do we enforce anywhere that `len(ins) == 1`?

File: trickle.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+// BeginBatch sets some batch variables on the node, and isn't forwarded.
+func (n *TrickleNode) BeginBatch(b edge.BeginBatchMessage) error {
+	n.dims = b.Dimensions()
+	n.name = b.Name()
+	return nil
+}
+
+// BatchPoint forwards a PointMessage
+func (n *TrickleNode) BatchPoint(bp edge.BatchPointMessage) error {
+	return n.outs[0].Collect(edge.NewPointMessage(
```
Comment (danxmoran) [inline trickle.go:40]: Same question with `len(outs)`

File: integrations/batcher_test.go [hunk 4]
```diff
@@ -2605,6 +2602,42 @@ data
...
+	var script = `
+	var data = batch
+	|query('SELECT value FROM "telegraf"."default"."cpu"')
+		.period(4s)
+		.every(4s)
+		.groupBy('host')
+data
+	|trickle()
+	|window().period(10s)
+	|httpOut('TestBatch_Trickle')`
```
Comment (danxmoran) [inline integrations/batcher_test.go:2615]: Checking my understanding: this test is showing that `trickle()` followed by `window` is a way to re-batch data? i.e. in this case, re-batching from 4s windows into 10s windows?

File: trickle.go (hunk 2 above)
Comment (docmerlin) [reply trickle.go:25]: Yes, in the attachment, in the tick code.

File: trickle.go (hunk 3 above)
Comment (docmerlin) [reply trickle.go:40]: Yes, in the attachment, in the tick code.  if len is not 1, we will have errors way before it gets to here.

**Review by danxmoran — APPROVED** [review]

**Pull request merged**